samples/net/mqtt_azure/src/main.c

```diff
@@ -0,0 +1,470 @@
+/*
+ * Copyright (c) 2019 Intel Corporation
+ *
+ * SPDX-License-Identifier: Apache-2.0
+ */
+
+#include <logging/log.h>
+LOG_MODULE_REGISTER(mqtt_azure, LOG_LEVEL_DBG);
+
+#include <zephyr.h>
+#include <net/socket.h>
+#include <net/mqtt.h>
+
+#include <sys/printk.h>
+#include <string.h>
+#include <errno.h>
+
+#include "config.h"
+#include "test_certs.h"
+
+/* Buffers for MQTT client. */
+static u8_t rx_buffer[APP_MQTT_BUFFER_SIZE];
+static u8_t tx_buffer[APP_MQTT_BUFFER_SIZE];
+
+/* The mqtt client struct */
+static struct mqtt_client client_ctx;
+
+/* MQTT Broker details. */
+static struct sockaddr_storage broker;
+
+#if defined(CONFIG_SOCKS)
+static struct sockaddr socks5_proxy;
+#endif
+
+/* Socket Poll */
+static struct pollfd fds[1];
+static int nfds;
+
+static bool mqtt_connected;
+
+static struct k_delayed_work pub_message;
+#if !defined(CONFIG_BOARD_QEMU_X86)
```

Comment:
I was thinking to avoid board specific code control initially. Problem is what if we want to support DHCPv4 with Qemu?
Actual issue is QEMU and Native posix doesn't generate L4 events.

Reply:
So it seems that connection manager malfunctions with qemu. In that case, I'd disable it in qemu config (`CONFIG_NET_CONNECTION_MANAGER`), and use this config to enable/disable features in the sample.

But it could be that I misunderstand something, perhaps @jukkar will provide some more input on this case.

Reply:
> So it seems that connection manager malfunctions with qemu. In that case, I'd disable it in qemu config (`CONFIG_NET_CONNECTION_MANAGER`), and use this config to enable/disable features in the sample.

Not like that. Real board you can plug/unplug which generates L4connected and L4 disconnected events. In qemu/native_posix, case is different.
CONFIG_NET_CONNECTION_MANAGE also doesn't help I think. Again if qemu want to support dhcpv4 server on host. Things might go wrong.

Reply:
I still think that even on `qemu_x86` we should get `NET_EVENT_L4_CONNECTED` event once at startup, even if connection state can not change during sample execution. But again, I might misunderstand something, would be good to hear from others as well.

Reply:
Ok

Reply:
Modified to use CONFIG_NET_DHCPV4. Removed board specific.